xpk.py

```diff
@@ -1839,6 +1847,37 @@ def setup_docker_image(args) -> tuple[int, str]:
   return 0, docker_image
 
 
+def create_node_selector(args, system) -> str:
+  if args.device_type in TpuUserFacingNameToSystemCharacteristics:
+    return """cloud.google.com/gke-tpu-accelerator: {gke_accelerator}
+                cloud.google.com/gke-tpu-topology: {topology}
+    """.format(gke_accelerator=system.gke_accelerator, topology=system.topology)
+  elif args.device_type in GpuUserFacingNameToSystemCharacteristics:
+    return """cloud.google.com/gke-accelerator: {gke_accelerator}
+                cloud.google.com/gce-machine-type: {gce_machine_type}
+    """.format(gke_accelerator=system.gke_accelerator,
+               gce_machine_type=system.gce_machine_type)
+  else:
+    raise ValueError("Unknown device type")
+
+
+def get_resource_type(args) -> str:
+  if args.device_type in TpuUserFacingNameToSystemCharacteristics:
+    return "google.com/tpu"
+  elif args.device_type in GpuUserFacingNameToSystemCharacteristics:
+    return "nvidia.com/gpu"
+  else:
+    raise ValueError("Unknown device type")
```

Comment:
I think we can organize these "keys" like 'google.com/tpu' and 'nvidia.com/gpu' into another class like SystemCharacteristics. This would be nice to organize the GPU / TPU keys together with a dictionary instead of in functions.

Something like:

```
class AcceleratorType(Enum):
    TPU = 1
    GPU = 2
    # CPU = 3, one day

@dataclass
class AcceleratorCharacteristics:
  device_type: str
  accelerator_name: int
  type: str

AcceleratorTypeToAcceleratorCharacteristics = {
     #tpu
      'AcceleratorType.TPU': AcceleratorCharacteristics(
      'google.com/tpu', 'cloud.google.com/gke-accelerator', ' cloud.google.com/gce-machine-type')
    ), ...
}

# Could add a new member in system_characteristics which would be used as the key

class SystemCharacteristics:
  topology: str
  vms_per_slice: int
  gke_accelerator: str
  gce_machine_type: str
  chips_per_vm: int
  accelerator_type: AcceleratorType
```

Reply:
I'm rewriting these into a dictionary as you suggested, but it seems use of functions is still necessary. This is because we access either `system.topology` or `system.gce_machine_type` based on accelerator type.

we can directly use the dictionary for resource_type since these values are more static

Reply:
please see new commit https://github.com/google/xpk/pull/49/commits/6f524936211ba20e205e6fb3cb159e510cb79013